### DISCUSSION TITLE
Chunk loading is not working

### DISCUSSION THREAD

**Discussion by KianBrose** [Minestom/Minestom#2140] [Help] [answered]
After running the debugger for a while i discovered that chunks were simply not loading even though I explicitly tried to load them, why?

![image](https://github.com/Minestom/Minestom/assets/62078964/bb45ca6a-a508-48dc-a7ec-f9cdc1b5400d)

**Comment by ZakShearman** [comment, marked as answer]
InstanceContainer#loadChunk returns a CompletableFuture: https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/instance/InstanceContainer.java#L240

You need to wait these completable futures (CompletableFuture.allOf might be useful for you here) for the chunks to be loaded.

**Reply by KianBrose** [reply]
This was it, thank you
